Makefile

```diff
@@ -58,7 +58,7 @@ lean3-source: mathbin-source
 	if [ ! -d "sources/lean/.git" ]; then \
 		cd sources && git clone https://github.com/leanprover-community/lean.git; \
 	fi
-	cd sources/lean && git clean -xfd && git checkout `cd ../mathlib && lean --version | sed -e "s/.*commit \([0-9a-f]*\).*/\1/"`
+	cd sources/lean && git clean -xfd && git fetch && git checkout `cd ../mathlib && lean --version | sed -e "s/.*commit \([0-9a-f]*\).*/\1/"`
```

Comment:
I think it would be better to fetch the exact commit, which means repeating `` ``cd ../mathlib && lean --version | sed -e "s/.*commit \([0-9a-f]*\).*/\1/"` `` or putting it in a bash variable

Reply:
I've pushed a commit that might do this.